app/attributes/attributecontroller.cpp

```diff
@@ -904,6 +919,56 @@ void AttributeController::recalculateDerivedItems( bool isFormValueChange, bool
     }
   }
 
+  // Evaluate form items editability
+  {
+    QMap<QUuid, std::shared_ptr<FormItem>>::iterator formItemsIterator = mFormItems.begin();
+    while ( formItemsIterator != mFormItems.end() )
+    {
+      std::shared_ptr<FormItem> item = formItemsIterator.value();
+      bool editable = item->isEditable();
+      QgsExpression exp = item->editableExpression();
+      exp.prepare( &expressionContext );
+
+      if ( exp.isValid() )
+      {
+        editable = exp.evaluate( &expressionContext ).toInt();
+      }
+
+      if ( item->isEditable() != editable )
+      {
+        item->setEditable( editable );
+        changedFormItems << item->id();
+      }
+
+      ++formItemsIterator;
+    }
+  }
+
+  // Evaluate form items alias/name
+  {
+    QMap<QUuid, std::shared_ptr<FormItem>>::iterator formItemsIterator = mFormItems.begin();
+    while ( formItemsIterator != mFormItems.end() )
+    {
+      std::shared_ptr<FormItem> item = formItemsIterator.value();
+      QString name = item->name();
```

Comment:
I'd add `if` here to check if there is any name expression set and if not, `continue` to the next item. The same for the editable state